test/Microsoft.AspNetCore.Cors.Test/CorsMiddlewareTests.cs

```diff
@@ -349,5 +354,39 @@ public class CorsMiddlewareTests
                 Assert.Equal("PUT", response.Headers.GetValues(CorsConstants.AccessControlAllowMethods).FirstOrDefault());
             }
         }
+
+        [Fact]
+        public async Task CorsRequest_AfterResponseCleared_SetsResponseHeaders()
+        {
+            // Arrange
+            var hostBuilder = new WebHostBuilder()
+                .Configure(app => {
+                    app.UseCors(builder =>
+                        builder.WithOrigins("http://localhost:5001")
+                            .WithMethods("PUT")
+                            .WithHeaders("Header1")
+                            .WithExposedHeaders("AllowedHeader"));
+                    app.Run(async context => {
```

Comment:
Would be good to mimic a similar scenario as the original issue (i.e to register a middleware _before_ Cors and which catches exceptions globally and clears the response headers and sends out a generic safe response)